### ISSUE TITLE
windows only commands should exit when not on windows

### ISSUE THREAD

**Issue by btm** [chef/knife-windows#184]
On OS X:
```
$ knife windows listener create --cert-thumbprint 1F3A70E2601FA1576BC4850ED2D7EF6587076423
sh: winrm: command not found
WinRM listener created
```

```
$ knife windows cert install winrmcert.b64
Enter given certificate's passphrase (empty for no passphrase):
Adding certificate to the Certificate Store...
sh: powershell.exe: command not found
Error adding the certificate. Use -VV option for details
```

1) we shouldn't even try to run winrm if we're not on windows, we should exit with an error.
2) even on windows, if the winrm command fails we shouldn't say that the listener was created

**Comment by btm** [comment]
https://github.com/chef/knife-windows/pull/181 appears to have fixed the second point, although I haven't tested it.

**Comment by siddheshwar-more** [comment]
@btm - We are looking into this

**Comment by btm** [comment]
@NimishaS can you verify if this was completed or not? I couldn't find a leankit card for it.

**Comment by NimishaS** [comment]
@btm, first point is not fixed yet. Created a leankit card for it.

**Comment by NimishaS** [comment]
Created https://github.com/chef/knife-windows/pull/250

**Comment by btm** [comment]
Fixed by #250 .
